pkg/fetcher/traits.go

```diff
@@ -14,3 +16,34 @@ type CompressedResource interface {
 	ReadCompressed() ([]byte, *ResourceError)
 	ReadCompressedGzip() ([]byte, *ResourceError)
 }
+
+type StringResource interface {
```

Comment:
Could you explain why you extracted these APIs in a new `StringResource` interface?

If I'm not mistaken, as we don't know if a `Resource` will contain plain text until we try to read it, you will need to implement this for all `Resource` structs?

As I can see there's a lot of code duplication for these implementations, I think these APIs don't fit Go very well. Maybe keeping only the top-level `ReadResourceAsString()` functions is enough? Alternatively I found this technique using a base embedded struct: https://go.dev/play/p/ByoUzfqOpn

Reply:
I've never liked the duplication either. I'll look into that technique, maybe I can get rid of those duplicate functions. If not, I'll just turn it into utility functions instead if you don't mind it not matching the mobile toolkit

Reply:
> If not, I'll just turn it into utility functions instead if you don't mind it not matching the mobile toolkit

Absolutely, I'm all in favor of keeping the APIs more idiomatic to the platform than exactly in sync between the toolkits.

Reply:
I've removed that interface and the repeated functions, the places where they're needed just use plain functions that perform the equivalent work